scenes/quests/lore_quests/quest_001/2_ink_combat/components/throwing_enemy/components/throwing_enemy.gd

```diff
@@ -180,6 +180,8 @@ func _on_timeout() -> void:
 	var player: Player = get_tree().get_first_node_in_group("player")
 	if not is_instance_valid(player):
 		return
+	if not allowed_labels:
+		return
 	_is_attacking = true
 	animated_sprite_2d.play(&"attack anticipation")
 	await animated_sprite_2d.animation_finished
```

Comment:
I don't think this will work because a few lines later we `await` the end of the `attack anticipation` animation. By that point the `allowed_labels` may have changed, if the player filled the last inkpot during the anticipation animation.

You could move this check down:

```suggestion
	_is_attacking = true
	animated_sprite_2d.play(&"attack anticipation")
	await animated_sprite_2d.animation_finished
	if not allowed_labels:
		return
```

I think that would work.

Reply:
Oh yes, I just figure out that myself when trying to fix something else. The `allowed_labels` may be empty after the `await`. Thanks.